reactor-core/src/main/java/reactor/util/Loggers.java

```diff
@@ -507,6 +526,39 @@ final String format(@Nullable String from, @Nullable Object... arguments){
 			return null;
 		}
 
+		private synchronized void logWithOptionalThrowable(String level, String format, Object... arguments) {
+			if(isLastElementThrowable(arguments)) {
+				int lastIndex = arguments.length - 1;
+				Object[] args = Arrays.copyOfRange(arguments, 0, lastIndex);
+				Throwable t = (Throwable) arguments[lastIndex];
+
+				this.log.format("[%s] (%s) %s\n", level.toUpperCase(), Thread.currentThread().getName(), format(format, args));
+				t.printStackTrace(this.log);
+				return;
+			}
+
+			this.log.format("[%s] (%s) %s\n", level.toUpperCase(), Thread.currentThread().getName(), format(format, arguments));
+		}
+
+		private synchronized void logErrorWithOptionalThrowable(String level, String format, Object... arguments) {
```

Comment:
This method is almost exactly the same as the one above, can you please unify them with a `PrintStream` as an argument?

Reply:
I have merged the methods into one, and modified it to accept a `PrintStream` as a parameter.